winterfell/webapp/src/js/claimConfirmCtrl.js

```diff
@@ -1,87 +1,87 @@
       }
     ];
 
-    // list of forms user has filled out
-    // TODO: placeholder forms for now
-    $scope.formsList = [
-      {
-        id: 'va_2723',
-        title: 'VA-27343 Intent to File',
-        summary: 'Something that blah blah holds your place in line when filing.'
-      }
-    ];
+    $scope.formsList = forms;
+    $scope.formsInfo = formTemplateService;
 
-    $scope.onClickAddAddress = function() {
-      return openModifyContactModal('address', $scope.mailingList);
-    };
+    console.log($scope);
 
-    $scope.onClickEditAddress = function(index) {
-      return openModifyContactModal('address', $scope.mailingList, index);
+    $scope.onClickEditAddress = function (index) {
+      openModifyContactModal();
     };
 
-    $scope.onClickAddEmail = function() {
-      return openModifyContactModal('email', $scope.emailList);
+    $scope.onClickEditEmail = function (index) {
+      openModifyEmailModal();
     };
 
-    $scope.onClickEditEmail = function(index) {
-      return openModifyContactModal('email', $scope.emailList, index);
-    };
+    function openModifyContactModal() {
+      var modalInstance = $uibModal.open({
+        scope: $scope,
+        controller: 'claimConfirmModifyAddressCtrl',
+        templateUrl: 'templates/modals/confirmClaimModifyAddress.html',
+        resolve: {
+          address: function () {
+            return _.cloneDeep($scope.userAddress);
+          }
+        }
+      });
+
+      modalInstance.result.then(function (address) {
+        console.log("new address", address);
+        $scope.userAddress = address;
+      }, function () {
+        console.log('modal-component dismissed at: ' + new Date());
+      });
+    }
 
-    function openModifyContactModal(forType, list, index) {
-      $scope.modalData = {
-        forType: forType,
-        list: list,
-        targetIndex: index >= 0 ? index : -1
-      };
-      return $uibModal.open({
+    function openModifyEmailModal() {
+      var modalInstance = $uibModal.open({
         scope: $scope,
-        controller: 'claimConfirmModifyContactCtrl',
-        templateUrl: 'templates/modals/confirmClaimModifyContact.html'
+        controller: 'claimConfirmModifyEmailCtrl',
+        templateUrl: 'templates/modals/confirmClaimModifyEmail.html',
+        resolve: {
+          email: function () {
+            return $scope.userEmail;
+          }
+        }
+      });
+
+      modalInstance.result.then(function (email) {
+        console.log("new email", email);
+        $scope.userEmail = email;
+      }, function () {
+        console.log('modal-component dismissed at: ' + new Date());
       });
     }
 
-    $scope.onClickConfirm = function() {
-      var claim = claimService.getIncompleteClaim();
-      if (!_.isEmpty(claim)) {
-        net.submitClaim($stateParams.claimId).then(function(resp) {
+    $scope.onClickConfirm = function () {
+      net.submitClaim($stateParams.claimId,
+        {
+          toAddress: $scope.vaAddress,
```

Comment:
I think having the service `sendLetter` function mail to one single address is a simple and most reusable interface. In our application logic we can easily call the `sendLetter` function twice to mail to two separate addresses. I agree I haven't implemented the "send copy" functionality, but I'd like to address that in a future PR because this one is getting way too large and its starting to block my progress because I have to rebase my stuff off of here all the time since master has gotten far behind.